bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -1868,8 +1905,58 @@ public static Image win32_new(Device device, int type, long handle, int nativeZo
 	return new Image(device, type, handle, nativeZoom);
 }
 
+private <T> Optional<T> getElementAtTargetSize(BiFunction<Integer, Integer, Optional<T>> elementForZoomProvider,
+		int targetWidth, int targetHeight) {
+	Rectangle originalSize = getBounds(100);
+	int originalWidth = originalSize.width;
+	int originalHeight = originalSize.height;
+
+	Optional<T> dataAtOriginalZoom = elementForZoomProvider.apply(targetWidth, targetHeight);
+	if (dataAtOriginalZoom.isPresent()) {
+		T data = dataAtOriginalZoom.get();
+		if (data instanceof String fileName) {
+			ImageData imageData = ImageDataLoader.loadByTargetSize(fileName, targetWidth, targetHeight);
+			if (imageData.width == targetWidth && imageData.height == targetHeight) {
+				return dataAtOriginalZoom;
+			}
+		} else if (data instanceof ImageData)
+			return dataAtOriginalZoom;
+	}
+
+	//TODO: zoom ausrechnen und dann den größeren nehmen
+	if (targetWidth >= originalWidth && targetWidth <= originalWidth * 1.5) {
+		Optional<T> dataAt150Percent = elementForZoomProvider.apply((int) Math.round(originalWidth * 1.5),
+				(int) Math.round(originalHeight * 1.5));
+		if (dataAt150Percent.isPresent()) {
+			return dataAt150Percent;
+		}
+	}
+
+	if (targetWidth > originalWidth) {
+		Optional<T> dataAt200Percent = elementForZoomProvider.apply(originalWidth * 2, originalHeight * 2);
+		if (dataAt200Percent.isPresent()) {
+			return dataAt200Percent;
+		}
+	}
+
+	if (targetWidth != originalWidth) {
+		Optional<T> dataAt100Percent = elementForZoomProvider.apply(originalWidth, originalHeight);
+		if (dataAt100Percent.isPresent()) {
+			return dataAt100Percent;
+		}
+	}
+
+	return Optional.empty();
+}
+
 private abstract class AbstractImageProviderWrapper {
 
+	protected boolean sizeAware = false;
+
+	public boolean isSizeAware (){
+		return this.sizeAware;
+	}
```

Comment:
I do not understand why this method is needed. Every image will be capable of providing a (potentially temporary) handle with the data represented at a specific size (in the worst case by just scaling the image data from the best fitting zoomed image). So there should be no need to any case distinction based on whether an image is in some way size aware or not.